File: src/JustSending/Data/Message.cs
using System;

namespace JustSending.Data
{
    public class Message
    {
        public string Id { get; set; }
        public int SessionMessageSequence { get; set; }

        public string SessionId { get; set; }

        public string SessionIdVerification { get; set; }
        /// <summary>
        /// Unique for each connected device in a particular session
        /// </summary>
        public string SocketConnectionId { get; set; }

        public string EncryptionPublicKeyAlias { get; set; }
        public string Text { get; set; }

        public DateTimeOffset DateSent { get; set; }
        public bool HasFile { get; set; }
        public long FileSizeBytes { get; set; }
        
        /// Indicate if this is system generated Message
        /// Not a user sent message
        public bool IsNotification { get; set; }
        public int DateSentEpoch { get; set; }
    }
}

File: rs/src/models/message.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, sqlx::FromRow)]
pub struct Message {
    #[sqlx(rename = "Id")]
    pub id: String,
    #[sqlx(rename = "SessionId")]
    pub session_id: String,
    #[sqlx(rename = "SessionIdVerification")]
    pub session_id_verification: Option<String>,
    #[sqlx(rename = "SocketConnectionId")]
    pub socket_connection_id: Option<String>,
    #[sqlx(rename = "EncryptionPublicKeyAlias")]
    pub encryption_public_key_alias: Option<String>,
    #[sqlx(rename = "Text")]
    pub text: String,
    #[sqlx(rename = "FileName")]
    pub file_name: Option<String>,
    #[sqlx(rename = "DateSent")]
    pub date_sent: String,
    #[sqlx(rename = "HasFile")]
    pub has_file: bool,
    #[sqlx(rename = "FileSizeBytes")]
    pub file_size_bytes: Option<i64>,
    #[sqlx(rename = "IsNotification")]
    pub is_notification: bool,
    #[sqlx(rename = "DateSentEpoch")]
    pub date_sent_epoch: i64,
}
